### PR TITLE
Export `elapseBlocking` to test binding, so slow sync work can be simulated such as a slow widget build

### LINKED ISSUE #112620: Export `elapseBlocking` to test binding, so slow sync work can be simulated such as a slow widget build
This is the issue for https://github.com/flutter/flutter/pull/112619

_(Copied from PR):_

There are needs to simulate sync heavy work, such as a slow widget build, in flutter widget tests. This method simply expose that.

As a remark, this cannot be replaced by `runAsync`. Consider the following example:

```dart
    testWidgets('can use to simulate slow build', (WidgetTester tester) async {
      final DateTime beforeTime = binding.clock.now();

      await tester.pumpWidget(Builder(builder: (_) {
        bool timerCalled = false;
        Timer.run(() => timerCalled = true);

        binding.elapseBlocking(const Duration(seconds: 1));

        // if we use `delayed` instead of `elapseBlocking`, such as
        // binding.delayed(const Duration(seconds: 1));
        // the timer will be called here. Surely, that violates how
        // a flutter widget build works
        expect(timerCalled, false);

        return Container();
      }));

      expect(binding.clock.now(), beforeTime.add(const Duration(seconds: 1)));
      binding.idle();
    });
```

As is discussed in the comments in the example, if we use `delayed`, timers will be fired when executing half of a build function, which is totally wrong.

### PR DESCRIPTION
**Pull request by fzyzcjy**
There are needs to simulate sync heavy work, such as a slow widget build, in flutter widget tests. This method simply expose that.

As a remark, this cannot be replaced by `runAsync`. Consider the following example:

```dart
    testWidgets('can use to simulate slow build', (WidgetTester tester) async {
      final DateTime beforeTime = binding.clock.now();

      await tester.pumpWidget(Builder(builder: (_) {
        bool timerCalled = false;
        Timer.run(() => timerCalled = true);

        binding.elapseBlocking(const Duration(seconds: 1));

        // if we use `delayed` instead of `elapseBlocking`, such as
        // binding.delayed(const Duration(seconds: 1));
        // the timer will be called here. Surely, that violates how
        // a flutter widget build works
        expect(timerCalled, false);

        return Container();
      }));

      expect(binding.clock.now(), beforeTime.add(const Duration(seconds: 1)));
      binding.idle();
    });
```

As is discussed in the comments in the example, if we use `delayed`, timers will be fired when executing half of a build function, which is totally wrong.

*List which issues are fixed by this PR. You must list at least one issue.*
Close https://github.com/flutter/flutter/issues/112620

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_test/test/bindings_test.dart [hunk 1]
```diff
@@ -17,7 +17,8 @@ import 'package:flutter_test/flutter_test.dart';
 import 'package:test_api/test_api.dart' as test_package;
 
 void main() {
```
Comment (jonahwilliams) [inline packages/flutter_test/test/bindings_test.dart:19]: Nit: leave the existing formatting for unrealted code

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_test/test/bindings_test.dart (hunk 1 above)
Comment (fzyzcjy) [reply packages/flutter_test/test/bindings_test.dart]: Done

**Review by pdblasi-google — APPROVED** [review]
LGTM

**Pull request merged**